tests/unit/acquisition/multi_objective/test_dominance.py

```diff
@@ -101,9 +108,40 @@
             ),
             tf.constant([0, 4, 1, 0, 1, 0]),
         ),
+        (
+            tf.zeros((0, 3)),
+            tf.zeros((0, 3)),
+            tf.ones((0,)),
+        ),
     ],
 )
-def test_dominated_sort(scores: tf.Tensor, pareto_set: tf.Tensor, dominance: tf.Tensor) -> None:
-    ret_pareto_set, ret_dominance = non_dominated(scores)
-    npt.assert_allclose(ret_pareto_set, pareto_set)
-    npt.assert_array_equal(ret_dominance, dominance)
+@pytest.mark.parametrize("compiler", [lambda f: f, tf.function])
+def test_dominated_sort(
+    compiler: Compiler, scores: tf.Tensor, pareto_set: tf.Tensor, dominance: tf.Tensor
+) -> None:
+    compiled_non_dominated = compiler(non_dominated)
+    ret_pareto_set, ret_nondominated = compiled_non_dominated(scores)
+    npt.assert_allclose(tf.sort(ret_pareto_set, 0), tf.sort(pareto_set, 0))
+    npt.assert_array_equal(ret_nondominated, dominance == 0)
+
+
+@pytest.mark.parametrize("num_objectives", [2, 4, 6])
+@pytest.mark.parametrize("compiler", [lambda f: f, tf.function])
+def test_dominated_sort_scales_ok(compiler: Compiler, num_objectives: int) -> None:
```

Comment:
Typo (mine originally)

```suggestion
def test_dominated_scales_ok(compiler: Compiler, num_objectives: int) -> None:
```